File: app/src/main/java/com/fivelove/viewmodel/MessagesViewModel.java
package com.fivelove.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.fivelove.App;
import com.fivelove.db.model.Message;
import com.fivelove.db.repository.MessagesRepository;

import java.util.List;

/**
 * Created by Nguyen Kim Khanh on 6/16/2020.
 */
public class MessagesViewModel extends AndroidViewModel {
    private MessagesRepository repository;
    private LiveData<List<Message>> allMessages;

    public MessagesViewModel(@NonNull App application) {
        super(application);
        repository = new MessagesRepository(application);
        allMessages = repository.getAllMessages();
    }


    public void insertMessage(Message message) {
        repository.insertMessage(message);
    }

    public void updateMessage(Message message) {
        repository.updateMessage(message);
    }

    public void deleteMessage(Message message) {
        repository.deleteMessage(message);
    }

    public LiveData<List<Message>> getAllMessages() {
        return allMessages;
    }


}

File: app/src/main/java/com/fivelove/viewmodel/MessagesViewModel.kt
package com.fivelove.viewmodel

import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import com.fivelove.App
import com.fivelove.db.model.Message
import com.fivelove.db.repository.MessagesRepository
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

/**
 * Created by Nguyen Kim Khanh on 6/16/2020.
 */
class MessagesViewModel(application: App) : AndroidViewModel(application) {
    private val repository: MessagesRepository = MessagesRepository(application)
    private val allMessages: LiveData<List<Message>>
    fun insertMessage(message: Message?) {
        viewModelScope.launch(Dispatchers.IO) {
            repository.insertMessage(message)
        }
    }

    fun updateMessage(message: Message?) {
        viewModelScope.launch(Dispatchers.IO) {
            repository.updateMessage(message)
        }
    }

    fun deleteMessage(message: Message?) {
        viewModelScope.launch(Dispatchers.IO) {
            repository.deleteMessage(message)
        }
    }

    init {
        allMessages = repository.allMessages
    }
}